translation-web/src/main/java/eu/europeana/api/translation/web/service/TranslationWebService.java

```diff
@@ -130,17 +135,98 @@ CachedTranslationService instantiateCachedTranslationService(boolean useCaching,
     }
   }
 
+  /*
+   * This method is used only for the purpose of eTranslation stress test and can be excluded afterwards
+   */
+  private void limitTextSizeForETranslationStressTest(TranslationRequest translationRequest, List<TranslationObj> translObjs) {
```

Comment:
this implementation looks overcomplicated, and it doesn't seem to take into account that we also append the delimiter

Reply:
It takes into account the delimiter while calculating the length of the joint texts (so the length is with the delimiter), to be below the limit of 5000.